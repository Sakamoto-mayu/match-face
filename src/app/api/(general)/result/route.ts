import { NextResponse } from "next/server";

export async function POST(request: Request) {
  try {
    const req = await request.json();
    const project_id = req.project_id;
    const response = await fetch(
      `${process.env.BE_URL}/answer_requests/${project_id}`,
      {
        cache: "no-cache",
      }
    );
    if (!response.ok) {
      throw new Error(`Error ${response.status}: ${response.statusText}`);
    }
    const data = await response.json();
    return NextResponse.json(data);
  } catch (error: any) {
    return NextResponse.json(
      { error: `Error: ${error.message}` },
      { status: 500 }
    );
  }
}
