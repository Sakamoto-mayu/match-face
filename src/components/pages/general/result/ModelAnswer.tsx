import { QuestionList } from "@/types/(general)/(other)/result/questions";

const ModelAnswerContent = ({
  questionList,
  index,
}: {
  questionList: QuestionList;
  index: number;
}) => {
  // 模範回答の配列
  const modelAnswers = questionList.map((question) =>
    question.answerList.filter((answer) => answer.modelAnswerFl === true)
  );

  return (
    <>
      <div id="modelAnswer" className="my-7 p-3 rounded-md bg-light-gray">
        <h3 className="text-xl mb-2 ">模範解答</h3>
        {modelAnswers[index][0].context}
      </div>
    </>
  );
};

export default ModelAnswerContent;
