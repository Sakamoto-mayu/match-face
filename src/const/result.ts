export const Answer = [
  {
    user_id: "user1",
    project_id: "1",
    question_id: "1",
    answer_date: new Date("2023-07-06"),
    content:
      "回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。",
  },
  {
    user_id: "user1",
    project_id: "1",
    question_id: "2",
    answer_date: new Date("2023-07-06"),
    content: "選択肢A",
  },
  {
    user_id: "user1",
    project_id: "1",
    question_id: "3",
    answer_date: new Date("2023-07-06"),
    content:
      "回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。",
  },
  {
    user_id: "user1",
    project_id: "2",
    question_id: "1",
    answer_date: new Date("2023-07-06"),
    content:
      "回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。",
  },
  {
    user_id: "user1",
    project_id: "2",
    question_id: "2",
    answer_date: new Date("2023-07-06"),
    content:
      "回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。",
  },
  {
    user_id: "user1",
    project_id: "2",
    question_id: "3",
    answer_date: new Date("2023-07-06"),
    content:
      "回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。",
  },
  {
    user_id: "user1",
    project_id: "3",
    question_id: "1",
    answer_date: new Date("2023-07-06"),
    content:
      "回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。",
  },
  {
    user_id: "user1",
    project_id: "3",
    question_id: "2",
    answer_date: new Date("2023-07-06"),
    content:
      "回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。",
  },
  {
    user_id: "user1",
    project_id: "3",
    question_id: "3",
    answer_date: new Date("2023-07-06"),
    content:
      "回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。",
  },
];

export const ModelAnswer = [
  {
    editor_id: "admin1",
    project_id: "1",
    question_id: "1",
    edit_date: new Date("2023-07-07"),
    content:
      "模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。",
  },
  {
    editor_id: "admin1",
    project_id: "1",
    question_id: "2",
    edit_date: new Date("2023-07-07"),
    content:
      "模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。",
  },
  {
    editor_id: "admin1",
    project_id: "1",
    question_id: "3",
    edit_date: new Date("2023-07-07"),
    content:
      "模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。",
  },
  {
    editor_id: "admin1",
    project_id: "2",
    question_id: "1",
    edit_date: new Date("2023-07-07"),
    content:
      "模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。",
  },
  {
    editor_id: "admin1",
    project_id: "2",
    question_id: "2",
    edit_date: new Date("2023-07-07"),
    content:
      "模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。",
  },
  {
    editor_id: "admin1",
    project_id: "2",
    question_id: "3",
    edit_date: new Date("2023-07-07"),
    content:
      "模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。",
  },
  {
    editor_id: "admin1",
    project_id: "3",
    question_id: "1",
    edit_date: new Date("2023-07-07"),
    content:
      "模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。",
  },
  {
    editor_id: "admin1",
    project_id: "3",
    question_id: "2",
    edit_date: new Date("2023-07-07"),
    content:
      "模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。",
  },
  {
    editor_id: "admin1",
    project_id: "3",
    question_id: "3",
    edit_date: new Date("2023-07-07"),
    content:
      "模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。",
  },
];

export const Comment = [
  {
    user_id: "user1",
    admin_id: "adminuser",
    project_id: "1",
    comment_date: new Date("2023-07-07"),
    content:
      "コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。",
  },
  {
    user_id: "user1",
    admin_id: "adminuser",
    project_id: "2",
    comment_date: new Date("2023-07-07"),
    content:
      "コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入りますコメント内容が入ります。",
  },
  {
    user_id: "user1",
    admin_id: "adminuser",
    project_id: "3",
    comment_date: new Date("2023-07-07"),
    content:
      "コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入りますコメント内容が入ります。",
  },
];

export const User = [
  {
    id: "adminuser",
    name: "管理者太郎",
    image_url: "/icon/human_icon.png",
  },
];
