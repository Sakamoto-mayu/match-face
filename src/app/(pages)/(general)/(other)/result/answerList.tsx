import { question } from "@/const/testing";
import { Answer } from "@/const/result";
import ModelAnswerContent from "./modelAnswer";

type Props = {
  project_id: number;
};

export const AnswerList = ({ project_id }: Props) => {
  const currentQuestion = question.filter(
    (question) => question.project_id === project_id
  );

  const currentAnswer = Answer.filter(
    (answer) => answer.project_id === project_id
  );

  return (
    <>
      {currentQuestion.map((question, index) => (
        <div key={index} className="py-5">
          <h3 className="text-xl">
            Q{index + 1}：{question.name}
          </h3>
          <p className="mt-2 mb-4">{question.content}</p>
          {question.type === "writing" ? (
            <>
              <div className="border border-gray p-3 mt-2 rounded-md shadow-md">
                <h3 className="mb-2 text-xl">あなたの回答</h3>
                {currentAnswer[index].content}
              </div>
            </>
          ) : (
            <>
              {question.choices?.map((choice, index) => (
                <div key={index}>
                  <input
                    id={`choice${index}`}
                    type="radio"
                    value={choice}
                    name={question.name}
                  />
                  <label htmlFor={`choice${index}`}>{choice}</label>
                </div>
              ))}
            </>
          )}
          <ModelAnswerContent
            project_id={project_id}
            question_id={question.id}
          />
        </div>
      ))}
    </>
  );
};
