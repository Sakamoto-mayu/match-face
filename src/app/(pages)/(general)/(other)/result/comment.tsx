import TextArea from "@/components/ui/TextArea";
import { Comment } from "@/const/result";
import Image from "next/image";

type Props = {
  user_id: string;
  project_id: number;
};

export const CommentContent = ({ user_id, project_id }: Props) => {
  const currentComment = Comment.filter(
    (comment) => comment.user_id === user_id && project_id === project_id
  );

  return (
    <>
      <h3 className="text-xl mb-3">〇〇からのコメント</h3>
      <div className="border border-black rounded-md p-3 flex justify-between ">
        <div className="w-1/6 flex justify-center items-center">
          <Image
            src={"/icon/human_icon.png"}
            alt="コメントしたユーザーのアイコン"
            width={70}
            height={70}
          />
        </div>
        <div className="w-5/6">{currentComment[0].content}</div>
      </div>
    </>
  );
};
