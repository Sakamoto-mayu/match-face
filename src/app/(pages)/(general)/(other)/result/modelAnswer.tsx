import { ModelAnswer } from "@/const/result";

type Props = {
  project_id: number;
  question_id: number;
};

const ModelAnswerContent = ({ project_id, question_id }: Props) => {
  const currentModelAnswer = ModelAnswer.filter(
    (modelAnswer) =>
      modelAnswer.project_id === project_id && question_id === question_id
  );
  return (
    <>
      <div className="my-7 p-3 rounded-md bg-gray-200">
        <h3 className="text-xl mb-2 ">模範解答</h3>
        {currentModelAnswer[0].content}
      </div>
    </>
  );
};

export default ModelAnswerContent;
