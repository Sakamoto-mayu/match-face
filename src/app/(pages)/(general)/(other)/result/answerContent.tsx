"use client";

import { project } from "@/const/testing";
import { AnswerList } from "./answerList";
import { CommentContent } from "./comment";
import OrangeButton from "@/components/ui/button/OrangeButton";

export const AnswerContent = ({ id }: { id: number }) => {
  const currentProject = project.filter((project) => project.id === id);

  return (
    <div className="w-1/2 py-5">
      <div className="flex justify-between items-center my-3">
        <h1 className="text-2xl">回答の確認</h1>
        <p>回答日時：</p>
      </div>
      <hr className="border border-black" />
      <h2 className="text-xl my-2">{currentProject[0].name}</h2>
      <p className="text-lg">{currentProject[0].project_detail}</p>
      <AnswerList id={id} />
      <CommentContent />
      <div className="flex justify-around my-20">
        <OrangeButton label="質問一覧へ" />
        <OrangeButton label="回答履歴一覧へ" />
      </div>
    </div>
  );
};
