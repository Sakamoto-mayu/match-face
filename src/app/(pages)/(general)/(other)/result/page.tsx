import { AnswerContent } from "./answerContent";

const ResultPage = () => {
  const data = {
    user_id: "user1",
    project_id: 1,
  };

  return (
    <div className="flex flex-col items-center">
      <AnswerContent user_id={data.user_id} project_id={data.project_id} />
    </div>
  );
};

export default ResultPage;
