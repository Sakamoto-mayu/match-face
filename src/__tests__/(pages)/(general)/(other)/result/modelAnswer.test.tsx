import ModelAnswerContent from "@/components/pages/general/result/ModelAnswer";
import { render, screen } from "@testing-library/react";
import { ModelAnswer } from "@/const/result";
import "@testing-library/jest-dom";

const testData = {
  project_id: "1",
  question_id: "1",
  modelAnswerContent: ModelAnswer[0].content,
};

describe("ModelAnswerContentコンポーネントのテスト", () => {
  test("模範回答が正しく表示される", () => {
    render(
      <ModelAnswerContent
        project_id={testData.project_id}
        question_id={testData.question_id}
      />
    );

    const heading = screen.getByRole("heading");
    expect(heading).toHaveTextContent("模範解答");

    const modelAnswer = document.querySelector("#modelAnswer");
    expect(modelAnswer).toHaveTextContent(testData.modelAnswerContent);
  });

  test("スナップショットテスト", () => {
    const content = render(
      <ModelAnswerContent
        project_id={testData.project_id}
        question_id={testData.question_id}
      />
    );
    expect(content).toMatchSnapshot();
  });
});
