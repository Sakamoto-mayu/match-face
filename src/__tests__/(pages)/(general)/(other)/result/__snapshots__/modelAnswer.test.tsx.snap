// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModelAnswerContentコンポーネントのテスト スナップショットテスト 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="my-7 p-3 rounded-md bg-gray-200"
        id="modelAnswer"
      >
        <h3
          class="text-xl mb-2 "
        >
          模範解答
        </h3>
        模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="my-7 p-3 rounded-md bg-gray-200"
      id="modelAnswer"
    >
      <h3
        class="text-xl mb-2 "
      >
        模範解答
      </h3>
      模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
